orchestration/task_dispatcher.py

```
@@ -982,64 +1124,136 @@ def create_dynamic_agent(self, agent_spec: dict) -> bool:
 """
 
             # Write prompt to file to avoid shell quoting issues
-            prompt_file = os.path.join("/tmp", f"agent_prompt_{agent_name}.txt")
+            prompt_file = os.path.join("/tmp", f"agent_prompt_{agent_token}.txt")
```

Ruff (0.13.3)

Probable insecure usage of temporary file or directory: "/tmp"

(S108)